### ISSUE TITLE
Screen reader does not read documentation link

### ISSUE THREAD

**Issue by RMacfarlane** [microsoft/vscode#89570]
Issue Type: <b>Bug</b>

Testing https://github.com/microsoft/vscode/issues/89320

Create an extension that uses the new diagnostic link API:
```ts
export function activate(context: vscode.ExtensionContext) {
	const fooErrors = vscode.languages.createDiagnosticCollection('foo');
	let disposable = vscode.commands.registerCommand('extension.helloWorld', () => {
		fooErrors.clear();
	
		const activeDocUri = vscode.window.activeTextEditor!.document.uri;
	
		fooErrors.set(activeDocUri, [
			{
				range: new vscode.Range(0, 0, 0, 5),
				message: 'You should always use semicolon;',
				source: 'eslint',
				code: '@typescript-eslint/semi',
				code2: {
					value: '@typescript-eslint/semi',
					link: vscode.Uri.parse('https://eslint.org/docs/rules/semi')
				},
				severity: vscode.DiagnosticSeverity.Error,
			},
			{
				range: new vscode.Range(1, 0, 1, 5),
				// message: 'Ensures that double quoted strings are passed to a localize call\nto provide proper strings for\ndifferent locales',
				message: 'Ensures that double\nquoted strings are passed to a localize call to provide proper strings for different locales',
				source: 'eslint',
				code2: {
					value: 'code-no-unexternalized-strings',
					link: vscode.Uri.parse('https://github.com/microsoft/tslint-microsoft-contrib/blob/master/src/noUnexternalizedStringsRule.ts')
				},
				severity: vscode.DiagnosticSeverity.Error,
			}
		]);
	});

	context.subscriptions.push(disposable);
}
```

Turn on NVDA.

In the problems panel, when navigating through the diagnostics, nothing gets read to indicate that there is a link that could be followed.

In the problems peek view (F8), the "value" text is repeated three times

VS Code version: Code - Insiders 1.42.0-insider (c4b6a6b537de49a734d0eedea06152ea37dfdb6a, 2020-01-28T06:36:30.779Z)

**Comment by sandy081** [comment]
I will take care of this.

**Comment by sandy081** [comment]
Actually in Problems view, when you tab on the message it takes to the link.

@isidorn Can you help me in understanding what is the right thing to do here. May be can we get validated with some accessibility expert?

**Comment by isidorn** [comment]
I just tried this out, and what is the actual link? 
From trying out @RMacfarlane sample what I see wrong is that you have this expand control to expand multi line elements. Please set the `aria-expanded` property to reflect if the multi line element is expanded.

Is the link no rendered for me, here's what I see
![Screenshot 2020-06-12 at 17 39 57](https://user-images.githubusercontent.com/1926584/84520373-c320b900-acd3-11ea-8029-0ceb30bf17ed.png)

fyi @jvesouza since I know you use the problems view and you might have additional feedback on it

**Comment by jvesouza** [comment]
When I access the list of problems and press the tab key, the focus is moved to the show fixes button. I press the enter key but nothing seems to happen.

**Comment by isidorn** [comment]
@jvesouza thanks, that sounds like a bug. @sandy081 can you reproduce this?

**Comment by sandy081** [comment]
I can't and enter key opens the quick fixes for me.

> Please set the aria-expanded property to reflect if the multi line element is expanded.

Sure

Please try with following extension sample

```ts
export function activate(context: vscode.ExtensionContext) {

	const collection = vscode.languages.createDiagnosticCollection('test');
	const uri = vscode.workspace.workspaceFolders![0].uri.with({ path: path.join(vscode.workspace.workspaceFolders![0].uri.path, 'package.json') });
	collection.set(uri, [createDiagnostic(uri, new vscode.Range(new vscode.Position(3, 4), new vscode.Position(3, 10)))]);

	vscode.window.onDidChangeActiveTextEditor(e => {
		if (vscode.window.activeTextEditor) {
			collection.set(vscode.window.activeTextEditor.document.uri, [createDiagnostic(vscode.window.activeTextEditor.document.uri, new vscode.Range(new vscode.Position(3, 5), new vscode.Position(3, 10)))]);
		}
	});
}

function createDiagnostic(uri: vscode.Uri, range: vscode.Range): vscode.Diagnostic {
	return {
		message: 'this is a sample error',
		range,
		severity: vscode.DiagnosticSeverity.Error,
		source: 'abcd',
		code: {
			value: '@typescript-eslint/semi',
			target: vscode.Uri.parse('https://eslint.org/docs/rules/semi')
		},
	};
}
```

**Comment by isidorn** [comment]
@sandy081 then it is probably fixed with our other improvements in the markers panel.
Feel free to close once you add the `aria-expanded`. Thanks!
